doc/rados/configuration/mclock-config-ref.rst

```diff
@@ -247,6 +247,129 @@ And that's it! You are ready to run workloads on the cluster and check if the
 QoS requirements are being met.
 
 
+Switching Between Built-in and Custom Profiles
+==============================================
+
+There may be situations requiring switching from a built-in profile to the
+*custom* profile and vice-versa. The following sections outline the steps to
+achieve the same.
+
+Steps to Switch From a Built-in to the Custom Profile
+-----------------------------------------------------
+
+The following command can be used to switch to the *custom* profile:
+
+  .. prompt:: bash #
+
+    ceph config set osd[.N] osd_mclock_profile custom
+
+For example, to change the profile to *custom* on the OSDs, the following
+command can be used:
+
+  .. prompt:: bash #
+
+    ceph config set osd osd_mclock_profile custom
+
+After switching to the *custom* profile, the desired mClock configuration
+option may be modified. For example, to change the client reservation IOPS
+allocation for a specific OSD (say osd.0), the following command can be used:
+
+  .. prompt:: bash #
+
+    ceph config set osd.0 osd_mclock_scheduler_client_res 3000
+
+.. important:: Care must be taken to change the reservations of other services like
+   recovery and background best effort accordingly to ensure that the sum of the
+   reservations do not exceed the maximum IOPS capacity of the OSD.
+
+.. tip::  The reservation and limit parameter allocations must be on a per shard
+   based on the type of backing device (HDD/SSD) under the OSD. See
+   :confval:`osd_op_num_shards_hdd` and :confval:`osd_op_num_shards_ssd` for
+   more details.
+
+Steps to Switch From the Custom Profile to a Built-in Profile
+-------------------------------------------------------------
+
+Switching from the *custom* profile to a built-in profile requires an
+additional intermediate step of removing the custom settings from the
+central config database for the changes to take effect.
+
+The following sequence of commands can be used to switch to a built-in profile:
+
+#. Set the desired built-in profile using:
+
+   .. prompt:: bash #
+
+     ceph config set osd[.N] <mClock Configuration Option>
+
+   For example, to set the built-in profile to ``high_client_ops`` on all OSDs,
+   run the following command:
+
+   .. prompt:: bash #
+
+     ceph config set osd osd_mclock_profile high_client_ops
+#. Remove the custom configuration settings from the ConfigMonitor database:
+
+   .. prompt:: bash #
+
+     ceph config rm osd[.N] <mClock Configuration Option>
+
+   For example, to remove the configuration option
+   :confval:`osd_mclock_scheduler_client_res` that was set on the OSDs in
+   previous section, run the following command:
+
+   .. prompt:: bash #
+
+     ceph config rm osd osd_mclock_scheduler_client_res
+#. At this point the configuration settings pertaining to ``high_client_ops``
+   should come into effect. This can be verified by using the
+   *ceph config show ...* command.
+
+Switch Temporarily Between mClock Profiles
```

Comment:
This section seems only for advanced users or for testing purposes. Let's add a cautionary note about it. The last thing we want is for users to shoot themselves in the foot :)

Reply:
^ this

Reply:
Sure, I added a warning note to highlight this.